src/main/java/me/karakelley/http/HttpConnectionHandler.java

```diff
@@ -0,0 +1,37 @@
+package me.karakelley.http;
+
+import me.karakelley.http.utility.BufferedLineReader;
+import me.karakelley.http.utility.LineReader;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.InputStreamReader;
+import java.io.PrintWriter;
+import java.net.Socket;
+import java.util.concurrent.CompletableFuture;
+
+class HttpConnectionHandler {
+  private final Socket clientSocket;
+  Logger logger = LoggerFactory.getLogger(HttpConnectionHandler.class);
+
+  public HttpConnectionHandler(Socket clientSocket) {
+    this.clientSocket = clientSocket;
+  }
+
+  public void start() {
+    CompletableFuture.runAsync(() -> {
+      try (LineReader reader = new BufferedLineReader(new InputStreamReader(clientSocket.getInputStream()));
+           PrintWriter out = new PrintWriter(clientSocket.getOutputStream())) {
+        Request newRequest = new Request(reader);
+        sendResponse(newRequest, out);
+      } catch (Exception e) {
+        logger.info(e.getMessage());
```

Comment:
Generally when you're logging exceptions it's best to just pass the entire exception...

```
logger.info("Ouch!", e);
```

The logging framework will genearlly take care of logging the message and the entire stack trace for you.